graphql/database/missions/createMission.js

```diff
@@ -0,0 +1,45 @@
+import { nanoid } from 'nanoid'
+import UserMissionModel from './UserMissionModel'
+import MissionModel from './MissionModel'
+import UserModel from '../users/UserModel'
+import {
+  getPermissionsOverride,
+  MISSIONS_OVERRIDE,
+} from '../../utils/permissions-overrides'
+
+const override = getPermissionsOverride(MISSIONS_OVERRIDE)
+/**
+ * @param {Object} userContext - The user context.
+ * @param {string} userId - The user's Id
+ * @param {string} squadName - the name of the squad being created
+ * @return {Promise<Object>}  A promise that resolves into an object containing a squad id
+ */
+
+export default async (userContext, userId, squadName) => {
+  const user = await UserModel.get(userContext, userId)
+  // user should not have been able to create a new mission
+  if (user.currentMissionId) {
+    throw new Error(
+      'attempting to create a new squad while still in an ongoing missions'
+    )
+  }
+  const missionId = nanoid(9)
+  const newMission = {
+    id: missionId,
+    squadName,
```

Comment:
Let's limit the length of this. Limit characters with validator.js (semi-sanitize)? Not sure, LMK what you think.

Reply:
sounds good.  It seems more important to do this on the front end so they can't even create a squad with character length over 30, and if they hack it and somehow make a longer squad name there aren't really any repercussions to our data storage, but might as well do it on the backend as well.  Will also do the xss filter